core/chainio/avs_writer.go

```diff
@@ -81,69 +82,112 @@ func NewAvsWriterFromConfig(baseConfig *config.BaseConfig, ecdsaConfig *config.E
-	txNonce := big.NewInt(int64(tx.Nonce()))
+	txNonce := big.NewInt(int64(sim_tx.Nonce()))
 	txOpts.Nonce = txNonce
-	txOpts.GasPrice = tx.GasPrice()
+	txOpts.GasPrice = sim_tx.GasPrice()
 	txOpts.NoSend = false
 	i := 0
 
+	var sentTxs []*types.Transaction
+
+	batchMerkleRootHashString := hex.EncodeToString(batchMerkleRoot[:])
+
 	respondToTaskV2Func := func() (*types.Receipt, error) {
 		gasPrice, err := utils.GetGasPriceRetryable(w.Client, w.ClientFallback)
 		if err != nil {
 			return nil, err
 		}
-
-		bumpedGasPrice := utils.CalculateGasPriceBumpBasedOnRetry(gasPrice, gasBumpPercentage, gasBumpIncrementalPercentage, i)
-		// new bumped gas price must be higher than the last one (this should hardly ever happen though)
-		if bumpedGasPrice.Cmp(txOpts.GasPrice) > 0 {
-			txOpts.GasPrice = bumpedGasPrice
+		previousTxGasPrice := txOpts.GasPrice
+		// minimum gas price required for the new transaction to ensure it's 10% higher than the previous one.
+		minimumGasPriceBump := utils.CalculateGasPriceBumpBasedOnRetry(previousTxGasPrice, 10, 0, 0)
+		// calculate the suggested bumped gas price based on the current gas price and retry iteration.
+		suggestedBumpedGasPrice := utils.CalculateGasPriceBumpBasedOnRetry(
+			gasPrice,
+			gasBumpPercentage,
+			gasBumpIncrementalPercentage,
+			i,
+		)
+		// check the new gas price is sufficiently bumped.
+		// if the suggested bump does not meet the minimum threshold, use a fallback calculation to slightly increment the previous gas price.
+		if suggestedBumpedGasPrice.Cmp(minimumGasPriceBump) > 0 {
+			txOpts.GasPrice = suggestedBumpedGasPrice
 		} else {
-			// bump the last tx gas price a little by `gasBumpIncrementalPercentage` to replace it.
-			txOpts.GasPrice = utils.CalculateGasPriceBumpBasedOnRetry(txOpts.GasPrice, gasBumpIncrementalPercentage, 0, 0)
+			txOpts.GasPrice = utils.CalculateGasPriceBumpBasedOnRetry(
+				previousTxGasPrice,
+				gasBumpPercentage,
+				gasBumpIncrementalPercentage,
+				i,
+			)
 		}
 
 		if i > 0 {
+			w.logger.Infof("Trying to get old sent transaction receipt before sending a new transaction", "merkle root", batchMerkleRootHashString)
+			for _, tx := range sentTxs {
+				receipt, _ := w.Client.TransactionReceipt(context.Background(), tx.Hash())
+				if receipt == nil {
+					receipt, _ = w.ClientFallback.TransactionReceipt(context.Background(), tx.Hash())
+					if receipt != nil {
+						w.checkIfAggregatorHadToPaidForBatcher(tx, batchIdentifierHash)
+						return receipt, nil
+					}
+				}
+			}
+			w.logger.Infof("Receipts for old transactions not found, will check if the batch state has been responded", "merkle root", batchMerkleRootHashString)
+			batchState, _ := w.BatchesStateRetryable(&bind.CallOpts{}, batchIdentifierHash)
+			if batchState.Responded {
+				w.logger.Infof("Batch state has been already responded", "merkle root", batchMerkleRootHashString)
+				return nil, nil
```

Comment:
a comment that nil,nil means OK would be good

Reply:
I've added a description in the function at the top that explains this behavior.